src/provider/provider_tracking.c

```diff
@@ -27,10 +28,11 @@ typedef struct tracker_value_t {
 
 static umf_result_t umfMemoryTrackerAdd(umf_memory_tracker_handle_t hTracker,
                                         umf_memory_pool_handle_t pool,
-                                        const void *ptr, size_t size) {
+                                        umf_ba_pool_t *ba_pool, const void *ptr,
```

Comment:
couldn't ba_pool be a part of umf_memory_tracker_t? Also, I think you missed the malloc in umfMemoryTrackerCreate()

Reply:
Done